ambry-network/src/main/java/com.github.ambry.network/NetworkClient.java

```diff
@@ -89,8 +96,13 @@ public NetworkClient(Selector selector, NetworkConfig networkConfig, int maxConn
       throw new IllegalStateException("The NetworkClient is closed.");
     }
     List<ResponseInfo> responseInfoList = new ArrayList<ResponseInfo>();
+    numPendingConnections.set(pendingRequests.size());
     for (RequestInfo requestInfo : requestInfos) {
-      pendingRequests.add(new RequestMetadata(time.milliseconds(), requestInfo));
+      ClientNetworkRequestMetrics clientNetworkRequestMetrics =
+          new ClientNetworkRequestMetrics(networkMetrics.requestQueueTime, networkMetrics.requestSendTime,
+              networkMetrics.requestSendTotalTime, 0);
```

Comment:
Time spent till now probably could be used in a better way - to track the time spent since the request was created. But we can do that later (I have added it to the backlog)